src/textual/await_complete.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from asyncio import Future, gather, wait
+from typing import Coroutine
+
+
+class AwaitComplete:
+    """An 'optionally-awaitable' object."""
+
+    _instances: list["AwaitComplete"] = []
+    """Track all active instances of AwaitComplete."""
+
+    def __init__(self, *coroutine: Coroutine) -> None:
+        """Create an AwaitComplete.
+
+        Args:
+            coroutine: One or more coroutines to execute.
+        """
+        self.coroutine = coroutine
+        AwaitComplete._instances.append(self)
+        self._future: Future = gather(*[coroutine for coroutine in self.coroutine])
```

Comment:
I'm surprised `gather` returns a `Future`. It doesn't appear to be documented.

Reply:
Yeah, it returns a "GatheringFuture" instance 😄